plugins/ssr/ssrelim.ml

```diff
@@ -494,7 +498,7 @@ let equality_inj l b id c =
   when msg := Pp.string_of_ppcmds s;
        !msg = "Not a projectable equality but a discriminable one." ||
        !msg = "Nothing to inject." ->
-    Feedback.msg_warning (Pp.str !msg);
+    warn_injection !msg;
```

Comment:
Maybe a good opportunity to clean up this code so we don't string_of_ppcmds everything.

In https://github.com/coq/coq/blob/55f4095fe3c366a9f310584a55e2dc0605e5409c/tactics/equality.ml#L1432 where the user_err is created, instead we can do `tclZERO NothngToInject` with a custome purpose `NothingToInject` exception with registered printer, and match it here without caring about UserError.

The "not a projectable equality" message seems not to be used anymore (I can't find anywhere raising it).

Reply:
Indeed the discriminable message changed in d439a29f2551c6c8de753d481e8b3e26c27d248e (8.5) then the error was replaced by a success in 94e8664074cfb987f8a63ca29c5436c861184b3a (#276, 8.7).

Reply:
I'll give it a try